### ISSUE TITLE
Building a Windows installer package from a MinGW package

### ISSUE THREAD

**Issue by s5bug** [msys2/MSYS2-packages#3042]
I'm working on an update to the PKGBUILD for ffmpeg, to update ffmpeg to 5.0.1 and enable quite a few more features. I'd like to be able to use this to build a package I can replace my existing Windows-side ffmpeg with.

![image](https://user-images.githubusercontent.com/7751296/174230753-f714cd4d-bc02-43ff-b75b-d6e158ee46aa.png)

Is this something Msys2 can do for me automatically?

**Comment by Biswa96** [comment]
Do you mean like a standalone installer like GIMP or Inkscape does from msys2/mingw packages? or a pacman pacakge .tar.zst file which can be installed with `pacman -U` command?

**Comment by s5bug** [comment]
`mingw-makepkg` already does the latter right? Currently, I'm meaning the former (or at least a `zip` that can be directly extracted to `C:\Program Files\ffmpeg` without issues).

**Comment by Biswa96** [comment]
> mingw-makepkg already does the latter right?

Yes.

> Currently, I'm meaning the former

I am not familiar with creating installer. Others may help with that part. I have seen multiple ways to do it. msys2 creates installer using this repository https://github.com/msys2/msys2-installer. Simple SFX file can also be utilized using this method https://superuser.com/questions/283698/create-7-zip-sfx-executable

**Comment by Biswa96** [comment]
Speaking of ffmpeg, the official ffmpeg page provide some link with simple 7zip files here https://ffmpeg.org/download.html#build-windows

**Comment by 1480c1** [comment]
Another option would be to get the resulting tar.zst (or whatever it is) and untar the specific folders you need, so something like
```bash
mkdir temp
cd temp
tar xf  ../*.tar.zst mingw64/{bin,include,lib}
```

and then zipping it, however, if your binaries requires other shared libraries, unless you are willing to mess around with ntldd, you could try something with pactree and bsdtar with something like
```bash
pactree -us mingw-w64-x86_64-ffmpeg |
    sort -u |
    pacman -Sddp - |
    xargs -n 1 sh -c  'curl -Ls "$1" | bsdtar -xv mingw64/bin mingw64/include mingw64/lib' _ 
```

and strip out the .exes that aren't from ffmpeg from the resulting `mingw64/bin` folder and zip up the insides of mingw64

**Comment by s5bug** [comment]
Yes, I'm aware that FFmpeg provides builds: my next goal is to build Kdenlive with support for Hardware Acceleration on my GPU. So, I need FFmpeg in a Mingw64 environment

**Comment by MehdiChinoune** [comment]
You could create a folder by installing ffmpeg by:
```
pacman -S --noconfirm ${MINGW_PACKAGE_PREFIX}-ffmpeg --root DESTINATION
```
This will pull and install all ffmpeg dependencies including their development files (static libs `.a`, import libs `.dll.a`, header files `.h`, docs, cmake files `.cmake`, pkgconfig files `.pc` ...),

**Comment by efa** [comment]
I made a bash packaging script for MSYS2's GTK apps, I think can be made generic to other packages simply removing GTK parts.
[gtk3msysPkg.tar.gz](https://github.com/user-attachments/files/17495595/gtk3msysPkg.tar.gz)
